### ISSUE TITLE
Cannot open newly created campaigns until FRED is restarted

### ISSUE THREAD

**Issue by Yarn366** [scp-fs2open/fs2open.github.com#1300]
If you create a new campaign in FRED and then try to reopen the campaign, FRED just throws an error saying "Can't open Campaign file!" The only way to be able to open the campaign again is restart FRED (not just the campaign editor). However, if a campaign with the same file name existed on startup, then FRED opens the new campaign without problems, whether you deleted the old file first or overwrote it.

It seems like FRED (or perhaps more accurately CFile) builds a list of all campaigns upon startup, and is able to open only those campaigns for the rest of the runtime, even if new campaigns are created.

This bug was introduced in commit 5a0bfff, the SDL Everywhere merge.

**Comment by asarium** [comment]
FSO builds a list of all files it knows at startup and then never touches that again. This issues can't be resolved without a lot of changes to how CFile works since it was never designed to handle a changing file system.

The reason this broke with the SDL merge is that the primary content root is now the mod folder in the user directory. FSO always looks in the first root if it can't find the specified file which is the reason why it worked before but since that root is now in the user directory, it can't find the newly saved file.

**Comment by Yarn366** [comment]
Well, FRED can open newly created _missions_ without a restart, so I would think the same can be done with campaigns without major changes to CFile.

Also, when you open a campaign, you're specifying an absolute path. Why can't FRED just use this path? It does so with missions, so why not campaigns?

**Comment by JohnAFernandez** [comment]
Going to try to see if I can handle this by 21.4

**Comment by jg18** [comment]
While this regression is certainly annoying/problematic and should be fixed when possible, I'm highly skeptical that it deserves Critical Priority, maybe not even High Priority, given that

- there's a straightforward workaround (restart FRED)
- modders probably don't create brand-new campaigns very often
- from @asarium 's comment (admittedly from March 2017) it sounds like this could be a lot of work to fix, and we're currently a small-ish team already stretched thin, so cost/benefit analysis doesn't look great to me

It's also telling that no modder has ever bumped this issue since it was created over *four years ago*.

Given the above, I really don't think it should be a release blocker. My $0.02.

**Comment by z64555** [comment]
The Critical column in the Bug Triage project is a WIP, awaiting for discussion and deliberation on the forums.  This issue was put in the column simply because of the `regression` tag, and I didn't look into the issue in depth before adding it.  There's merits to prioritizing regressions as a high or critical item so as to provide long-term support for mods that depend on (or rather broken by) regressions.

Still the fact that this issue hasn't been addressed shows that its either a low priority item or that players haven't complained enough about it

**Comment by JohnAFernandez** [comment]
Just a note, everytime I try to build FRED on this computer with any changes, it fails.

**Comment by MjnMixael** [comment]
Just tested this on the latest nightly and was able to open a newly campaign without restarting FRED.

**Comment by Goober5000** [comment]
This was probably fixed by #4885
